### ISSUE TITLE
Override added using macro annotation does not get the expected bridge because the symbol isn't entered in the class decls

### ISSUE THREAD

**Issue by smarter** [lampepfl/dotty#18806]
## Compiler version

Latest main: 58810fd00df8b50b67d64f5369ffc384c9bdef60

## Minimized code

`Macro_1.scala`:
```Scala
import scala.annotation.{experimental, MacroAnnotation}
import scala.quoted._

@experimental
class gen extends MacroAnnotation:
  def transform(using Quotes)(tree: quotes.reflect.Definition): List[quotes.reflect.Definition] =
    import quotes.reflect._
    tree match
      case ClassDef(name, ctr, parents, self, body) =>
        val cls = tree.symbol
        // val meth = cls.methodMember("foo").head
        // val fooTpe = cls.typeRef.memberType(meth)

        val overrideTpe =  MethodType(Nil)(_ => Nil, _ => defn.StringClass.typeRef)

        val fooOverrideSym = Symbol.newMethod(cls, "foo", overrideTpe, Flags.Override, Symbol.noSymbol)

        val fooDef = DefDef(fooOverrideSym, _ => Some(Literal(StringConstant("hi"))))

        val newClassDef = ClassDef.copy(tree)(name, ctr, parents, self, fooDef :: body)
        List(newClassDef)
      case _ =>
        report.error("Annotation only supports `class`")
        List(tree)
```
`Test_2.scala`:
```scala
class Base:
  def foo(): Object = ???

@gen
class Sub extends Base
// > override def foo(): String = "hi"

@main def Test(): Unit =
  val sub = new Sub
  println(sub.foo())
```

## Output

```scala
> scalac -Xcheck-macros Macro_1.scala Test_2.scala
> scala Test
Exception in thread "main" scala.NotImplementedError: an implementation is missing
        at scala.Predef$.$qmark$qmark$qmark(Predef.scala:344)
        at Base.foo(Test_2.scala:2)
        at Test_2$package$.Test(Test_2.scala:14)
        at Test.main(Test_2.scala:12)
```

## Expectation

```scala
> scala Test
hi
```

## Analysis

The output from `-Xprint:erasure` contains:
```scala
  class Sub() extends Base() {
    override def foo(): String = "hi"
  }
```

Notice that the bridge to `def foo(): Object` is missing. As far as I can tell the issue is that the overridden symbol is never entered into the decls of the enclosing class, so the bridge-generation logic does not even know it exists. Surprisingly -Ycheck:all does not catch this mismatch between decls and trees.

**Comment by hamzaremmal** [comment]
@nicolasstucki Should I take over this issue ?

**Comment by hamzaremmal** [comment]
Fixed in #18826
